src/amp-story-player/amp-story-player-impl.js

```diff
@@ -705,17 +727,22 @@ export class AmpStoryPlayer {
    * @param {number} storyDelta
    */
   go(storyDelta) {
-    if (
-      this.currentIdx_ + storyDelta >= this.stories_.length ||
-      this.currentIdx_ + storyDelta < 0
-    ) {
+    if (storyDelta === 0) {
+      return;
+    }
+    if (!this.circularWrapping_() && this.isIndexOutofBounds_(storyDelta)) {
       throw new Error('Out of Story range.');
     }
-    if (storyDelta === 0) {
+    if (!this.isIndexOutofBounds_(storyDelta)) {
```

Comment:
We can simplify from here all the way down this method like this:

```suggestion
   let currentStory;
   const newIdx = this.currentIdx_ + storyDelta;
    if (!this.isIndexOutofBounds_(newIdx)) {
      currentStory = this.stories_[newIdx];
    } else {
      currentStory = storyDelta > 0
        ? this.stories_[newIdx % this.stories_.length]
        : this.stories_[(newIdx % this.stories_.length + this.stories_.length) % this.stories_.length];
    }
    this.show(currentStory.href);
    this.signalNavigation_();
```

I set up a codepen to test the algorithm, you can play around with it here: https://codepen.io/maenrique/pen/qBbGymX?editors=0012